### PR TITLE
Fix inconsistencies with UserId

### LINKED ISSUE #626: appletOE IApplicationFunctions: 26 (GetSaveDataSize) not implemented
IApplicationFunctions GetSaveDataSize is not implemented.

## Service description:
[Switchbrew](https://switchbrew.org/wiki/Applet_Manager_services#IApplicationFunctions) | [Reswitched](https://reswitched.github.io/SwIPC/ifaces.html#nn::am::service::IApplicationFunctions(26))

```GetSaveDataSize(u8, nn::account::Uid) -> (i64, i64)```

Unknown.

## Required by:
- https://github.com/Ryujinx/Ryujinx-Games-List/issues/123
- https://github.com/Ryujinx/Ryujinx-Games-List/issues/136

### PR DESCRIPTION
**Pull request by Thog**
The account user id isn't an UUID. This PR adds a new UserId type with
the correct value ordering to avoid mismatch with LibHac's Uid. This also fix
an hardcoded value of the UserId.

As the userid has been invalid for quite some time (and to avoid forcing
users to their recreate saves), the userid has been changed to "00000000000000010000000000000000".

Also implement a stub for IApplicationFunctions::GetSaveDataSize. (see
the sources for the reason of the stub)

Fix #626

### CODE REVIEW

File: Ryujinx.HLE/HOS/Services/Am/AppletOE/ApplicationProxyService/ApplicationProxy/IApplicationFunctions.cs [hunk 1]
```diff
@@ -108,6 +110,23 @@ public ResultCode GetDisplayVersion(ServiceCtx context)
             return ResultCode.Success;
         }
 
+        // GetSaveDataSize(u8, nn::account::Uid) -> (u64, u64)
+        [Command(26)] // 3.0.0+
+        public ResultCode GetSaveDataSize(ServiceCtx context)
+        {
+            SaveDataType saveDataType = (SaveDataType)context.RequestData.ReadByte();
+            context.RequestData.BaseStream.Position += 7;
```
Comment (jduncanator) [inline Ryujinx.HLE/HOS/Services/Am/AppletOE/ApplicationProxyService/ApplicationProxy/IApplicationFunctions.cs]: Better to use `Seek()`?

File: Ryujinx.HLE/HOS/Services/Account/Acc/IAccountServiceForApplication.cs [hunk 2]
```diff
@@ -75,8 +75,8 @@ private ResultCode WriteUserList(ServiceCtx context, IEnumerable<UserProfile> pr
                     break;
                 }
 
-                context.Memory.WriteInt64(outputPosition + (long)offset,     userProfile.UserId.Low);
-                context.Memory.WriteInt64(outputPosition + (long)offset + 8, userProfile.UserId.High);
+                context.Memory.WriteInt64(outputPosition + (long)offset,         userProfile.UserId.High);
```
Comment (AcK77) [inline Ryujinx.HLE/HOS/Services/Account/Acc/IAccountServiceForApplication.cs]: There is an extra space here before `userProfile.UserId.High` and same for the line after.

File: Ryujinx.HLE/HOS/Services/Am/AppletOE/ApplicationProxyService/ApplicationProxy/IApplicationFunctions.cs [hunk 3]
```diff
@@ -39,7 +41,7 @@ public ResultCode PopLaunchParameter(ServiceCtx context)
         // EnsureSaveData(nn::account::Uid) -> u64
         public ResultCode EnsureSaveData(ServiceCtx context)
         {
-            Uid     userId  = context.RequestData.ReadStruct<Uid>();
+            Uid userId      = context.RequestData.ReadStruct<AccountUid>().ToLibHacUid();
```
Comment (AcK77) [inline Ryujinx.HLE/HOS/Services/Am/AppletOE/ApplicationProxyService/ApplicationProxy/IApplicationFunctions.cs]: If you align the assign then align the var name too

**Comment by Thog** [comment]
Ready for reviews

**Pull request merged**